src/os/bluestore/BlueFS.cc

```diff
@@ -1244,7 +1244,7 @@ void BlueFS::_compact_log_async(std::unique_lock<std::mutex>& l)
 	   << std::dec << dendl;
 
   // allocate
-  int r = _allocate(BlueFS::BDEV_DB, new_log_jump_to,
+  int r = _allocate(log_file->fnode.prefer_bdev, new_log_jump_to,
```

Comment:
My thinking here is that the async compaction isn't blocking, so it isn't latency sensitive.. and the wal device will generally only be big enough for the rocksdb .log files.  Since this is only ever read on startup, I don't think there's much value in putting it on the wal device.